### ISSUE TITLE
`claim.Messages()` is closed when two subscriber are subscribing to empty topic

### ISSUE THREAD

**Issue by roblaszczak** [Shopify/sarama#1564]
##### Versions

*Please specify real version numbers or git SHAs, not just "Latest" since that changes fairly regularly.*

| Sarama | Kafka | Go |
|--------|-------|----|
| v1.24.1 | `confluentinc/cp-kafka:5.3.1`  | `go1.13.5`  |
| d51425453712462fff8129cfd34cead25d67f779 | `confluentinc/cp-kafka:5.3.1`  | `go1.13.5`  |

##### Configuration

What configuration values are you using for Sarama and Kafka?
``` go
package main

import (
	"context"
	"fmt"

	"github.com/Shopify/sarama"
	"github.com/davecgh/go-spew/spew"
)

func main() {
	config := sarama.NewConfig()
	config.Version = sarama.V1_0_0_0
	config.Consumer.Return.Errors = true

	client, err := sarama.NewClient([]string{"localhost:9092"}, config)
	if err != nil {
		panic(err)
	}

	for {
		fmt.Println("starting sub")
		group, err := sarama.NewConsumerGroupFromClient("new_group", client)
		if err != nil {
			panic(err)
		}

		handler := consumerGroupHandler{}

		err = group.Consume(context.Background(), []string{"empty_topic"}, handler)
		if err != nil {
			panic(err)
		}

		err = group.Close()
		if err != nil {
			panic(err)
		}

		fmt.Println("consume closed, consuming again")
	}
}

type consumerGroupHandler struct {
}

func (consumerGroupHandler) Setup(_ sarama.ConsumerGroupSession) error { return nil }

func (consumerGroupHandler) Cleanup(_ sarama.ConsumerGroupSession) error { return nil }

func (h consumerGroupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
	kafkaMessages := claim.Messages()

	fmt.Println("Consume claimed")

	for {
		select {
		case kafkaMsg, ok := <-kafkaMessages:
			if !ok {
				fmt.Println("kafkaMessages is closed, stopping consumerGroupHandler")
				return nil
			}
			spew.Dump(kafkaMsg.Headers)
		}
	}
}

```

##### Logs

When filing an issue please provide logs from Sarama and Kafka if at all
possible. You can set `sarama.Logger` to a `log.Logger` to capture Sarama debug
output.

<details><summary>logs: CLICK ME</summary>
<p>

terminal 1
```
$ go run main.go

starting sub
consume closed, consuming again
starting sub
Consume claimed
kafkaMessages is closed, stopping consumerGroupHandler
consume closed, consuming again
starting sub
Consume claimed
kafkaMessages is closed, stopping consumerGroupHandler
consume closed, consuming again
starting sub
```
terminal 2
```
$go run main.go

starting sub
Consume claimed
kafkaMessages is closed, stopping consumerGroupHandler
consume closed, consuming again
starting sub
Consume claimed
kafkaMessages is closed, stopping consumerGroupHandler
consume closed, consuming again
starting sub
consume closed, consuming again
starting sub
Consume claimed
```

</p>
</details>

##### Problem Description

If subscribed topic is empty and `Consume()` is called from two consumers, `Consume()` constantly stops to work because `claim.Messages()` is closed.
After the first message arrives, `claim.Messages()` stops to be closed.

**Comment by Apologiz** [comment]
I have exactly the same problem. I debugged the library and found the reason.

consumerGroupSession has a method: heartbeatLoop() in which an error is returned: **RebalanceInProgressCode** (27). 

description: [Returned in heartbeat requests when the coordinator has begun rebalancing the group. This indicates to the client that it should rejoin the group](https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-ErrorCodes).

I solved the problem on the application side:
```golang
for {
	select {
	case <-ctx.Done():
		return ctx.Err()
	default:
		err := group.Consume(ctx, topics, &consumer)
		log.Info("group.Consume", zap.Error(err))
	}
}
```

@roblaszczak, I hope this solves your problem.
